sources/cg_c.c

```diff
@@ -916,7 +916,13 @@ static void cg_store(charbuf *output, CSTR var, sem_t sem_type_var, sem_t sem_ty
   CHARBUF_OPEN(adjusted_value);
   CG_BEGIN_ADJUST_FOR_OUTARG(var, sem_type_var);
 
-  // Most types convert correctly with no help, the C compiler will convert.  This is not true for bool.
+  // Normalize floats and bools for storage
```

Comment:
We may need to do this in other places too but for now I fixed the biggest offender -- assignment.

Note that only the latest clang complains about the missing cast.